### PR TITLE
Feature/alert structure

### PR DESCRIPTION
**Pull request by GabrielMoreschi**
Foram adicionados o filtro por TI, CR e Data.
A tabela está estruturada e condicionada a requisição retornar dados, que atualmente está implementada porém sem retorno. Aguardando o backend.

### CODE REVIEW

**Comment by GabrielMoreschi** [comment]
[ALERTA URGENTE - Criação do módulo de mapa e tabela](https://trello.com/c/dwUBKSVC/20-alerta-urgente-cria%C3%A7%C3%A3o-do-m%C3%B3dulo-de-mapa-e-tabela)

File: components/alert-urg/AlertFilter.vue [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            if (cr) this.$store.dispatch('alert-urg/getTiOptions', cr)
+            else this.filters.ti = null
+        },
+
+        search() {
+            this.setFilters(this.filters)
+            this.$emit('onSearch')
+        },
+        ...mapMutations('alert-urg', ['setFilters']),
+        ...mapActions('alert-urg', ['getFilterOptions', 'downloadGeoJson']),
```
Comment (dagnaldo) [inline components/alert-urg/AlertFilter.vue:251]: urgent-alerts

File: components/alert-urg/AlertFilter.vue [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        populateTiOptions(cr) {
+            if (cr) this.$store.dispatch('alert-urg/getTiOptions', cr)
+            else this.filters.ti = null
+        },
+
+        search() {
+            this.setFilters(this.filters)
+            this.$emit('onSearch')
+        },
+        ...mapMutations('alert-urg', ['setFilters']),
```
Comment (dagnaldo) [inline components/alert-urg/AlertFilter.vue:250]: urgent-alerts

File: store/alert-urg.js [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    },
+
+    setLoadingFeatures(state, payload) {
+        state.isLoadingFeatures = payload
+    },
+
+    clearFeatures(state) {
+        state.features = null
+    },
+    setLoadingCSV(state, payload) {
```
Comment (dagnaldo) [inline store/alert-urg.js:72]: Missing whitespace between functions

File: store/alert-urg.js [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+        const tis = await this.$api.$get('funai/ti/', { params })
+
+        if (tis)
+            commit('setFilterOptions', {
+                ...state.filterOptions,
+                tiFilters: tis.sort((a, b) => a.no_ti > b.no_ti),
+            })
+    },
+    async getDataTable({ commit, state, rootGetters }) {
```
Comment (dagnaldo) [inline store/alert-urg.js:199]: Missing blank line between functions

**Review by dagnaldo — CHANGES_REQUESTED** [review]
Check the variables, store and component names to work with more descriptive names.

TIP: functions have to be separated by blank line.
TIP2: Centralize CR and TI requests in a store (It will help you to write less code)

File: pages/index.vue [hunk 5]
```diff
@@ -93,7 +93,7 @@
         "support-fire-tab": "Risco de Fogo e Focos de Calor",
         "landuse-tab": "Uso e Ocupação do Solo",
         "compare-tab": "Comparar imagens",
-        "algorithms-tab": "Algoritmos",
+        "alert-urg-tab": "Alerta Urgente",
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply pages/index.vue]: Resolved

File: pages/index.vue [hunk 6]
```diff
@@ -80,7 +80,7 @@
         "support-fire-tab": "Fire Hazard and Hot Spots",
         "landuse-tab": "Land Use And Occupation",
         "compare-tab": "Compare images",
-        "algorithms-tab": "Algorithms",
+        "alert-urg-tab": "Alert Urgent",
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply pages/index.vue]: Resolved

File: components/alert-urg/AlertFilter.vue [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+                this.$store.commit('alert-urg/setOpacity', value)
+            },
+        },
+
+        heatMap: {
+            get() {
+                return this.$store.state.alert - urg.heatMap
+            },
+            set(value) {
+                this.$store.commit('alert-urg/setHeatMap', value)
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/alert-urg/AlertFilter.vue]: Resolved

File: components/alert-urg/AlertFilter.vue [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+            this.populateTiOptions(value)
+        },
+    },
+    computed: {
+        opacity: {
+            get() {
+                return this.$store.state.alert - urg.opacity
+            },
+            set(value) {
+                this.$store.commit('alert-urg/setOpacity', value)
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/alert-urg/AlertFilter.vue]: Resolved

File: components/alert-urg/AlertFilter.vue (hunk 1 above)
Comment (GabrielMoreschi) [reply components/alert-urg/AlertFilter.vue]: Resolved

File: components/alert-urg/AlertFilter.vue [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+        ]),
+    },
+
+    mounted() {
+        this.getFilterOptions()
+    },
+
+    methods: {
+        populateTiOptions(cr) {
+            if (cr) this.$store.dispatch('alert-urg/getTiOptions', cr)
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply components/alert-urg/AlertFilter.vue]: Resolved

File: pages/index.vue [hunk 10]
```diff
@@ -164,11 +164,11 @@ export default {
                     icon: 'mdi-sprout',
                     route: '/land-use',
                 },
-                // {
-                //     name: this.$t('algorithms-tab'),
-                //     icon: 'mdi-function-variant',
-                //     route: '/algorithms',
-                // },
+                {
+                    name: this.$t('alert-urg-tab'),
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply pages/index.vue]: Resolved

File: store/alert-urg.js (hunk 3 above)
Comment (GabrielMoreschi) [reply store/alert-urg.js]: Resolved

File: store/alert-urg.js [hunk 11]
```diff
@@ -0,0 +1,343 @@
...
+        state.isLoadingFeatures = payload
+    },
+
+    clearFeatures(state) {
+        state.features = null
+    },
+    setLoadingCSV(state, payload) {
+        state.isLoadingCSV = payload
+    },
+    setLoadingGeoJson(state, payload) {
```
[the remark this answers is not in the document]
Comment (GabrielMoreschi) [reply store/alert-urg.js]: Resolved

File: store/alert-urg.js (hunk 4 above)
Comment (GabrielMoreschi) [reply store/alert-urg.js]: Resolved

File: components/alert-urg/AlertFilter.vue [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+                    .format('YYYY-MM-DD'),
+                endDate: this.$moment().format('YYYY-MM-DD'),
+                cr: null,
+                ti: null,
+            },
+            isLoadingTotal: false,
+            legendData: legend,
+        }
+    },
+    watch: {
```
Comment (dagnaldo) [inline components/alert-urg/AlertFilter.vue:202]: Add blank line.

File: components/alert-urg/AlertFilter.vue [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+        'filters.cr'(value) {
+            this.populateTiOptions(value)
+        },
+    },
+    computed: {
```
Comment (dagnaldo) [inline components/alert-urg/AlertFilter.vue:207]: Add blank line.

File: pages/index.vue [hunk 14]
```diff
@@ -80,7 +80,7 @@
         "support-fire-tab": "Fire Hazard and Hot Spots",
         "landuse-tab": "Land Use And Occupation",
         "compare-tab": "Compare images",
-        "algorithms-tab": "Algorithms",
+        "urgent-alerts-tab": "Alert Urgent",
```
Comment (dagnaldo) [inline pages/index.vue:83]: Urgent Alert

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: pages/index.vue (hunk 14 above)
Comment (GabrielMoreschi) [reply pages/index.vue]: Resolved

File: components/alert-urg/AlertFilter.vue (hunk 13 above)
Comment (GabrielMoreschi) [reply components/alert-urg/AlertFilter.vue:193]: Resolved

File: components/alert-urg/AlertFilter.vue (hunk 12 above)
Comment (GabrielMoreschi) [reply components/alert-urg/AlertFilter.vue:187]: Resolved

**Review by dagnaldo — APPROVED** [review]
Well well, sir.
better late than never...

**Pull request merged**